drivers/hmis/app/graphql/types/base_audit_event.rb

```diff
@@ -103,9 +104,24 @@ def user
       # This happens when a Job updates records, which we should display as System changes.
       return User.system_user if object.whodunnit == 'unauthenticated'
 
-      # If user was impersonating, return the true user only. If someone performed the action while impersonating, whodunnit stores as '1 as 2' where 1 is the true user.
-      user_id = object.whodunnit.sub(/ as [0-9]+$/, '')
-      return unless user_id.to_i.to_s == user_id
+      user_id = [
+        # User user_id if available
+        object.user_id,
+        # Otherwise use whodunnit
+        object.whodunnit&.match?(/^\d+$/) ? object.whodunnit : nil,
+        object.whodunnit&.match?(whodunnit_impersonator_pattern) ? object.whodunnit.sub(whodunnit_impersonator_pattern, '\2') : nil,
```

Comment:
Can we move this logic onto the `GrdaWarehouse::Version` model? It's not really specific to GraphQL. That way, we can also use it in the `load_last_user_from_versions` method, since that also has some logic for looking for a numeric whodunnit.

For example on `Version` model:
`def clean_user_id` - pulls from `user_id` or `whodunnit` as appropriate, ensures numeric id
`def clean_true_user_id` - pulls from `true_user_id ` or `whodunnit` as appropriate, ensures numeric id